platform/darwin/src/MGLLight.h

```diff
@@ -0,0 +1,125 @@
+#import <CoreGraphics/CoreGraphics.h>
+
+#import "MGLFoundation.h"
+#import "MGLStyleValue.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+/** Options to specify extruded geometries are lit relative to the map or viewport. */
+typedef NS_ENUM(NSUInteger, MGLLightAnchor) {
+    /** The position of the light source is aligned to the rotation of the map. */
+    MGLLightAnchorMap,
+    /** The position of the light source is aligned to the rotation of the viewport. */
+    MGLLightAnchorViewport
+};
+
+/**
+ A structure containing information about the position of the light source
+ relative to lit geometries.
+ */
+typedef struct MGLSphericalPosition {
+    /** Distance from the center of the base of an object to its light. */
+    CGFloat radial;
+    /** Position of the light relative to 0° (0° when `MGLLight.anchor` is set to viewport corresponds
+     to the top of the viewport, or 0° when `MGLLight.anchor` is set to map corresponds to due north,
+     and degrees proceed clockwise). */
+    CGFloat azimuthal;
+    /** Indicates the height of the light (from 0°, directly above, to 180°, directly below). */
+    CGFloat polar;
```

Comment:
Oh, actually, based on the documentation, this field appears to be measured in degrees as well. It isn't clear to me whether it goes clockwise; if it does, then the appropriate type would be CLLocationDirection.